test/ContentValidationListenerTest.php

```diff
@@ -2338,4 +2338,118 @@ public function testEventNameShouldBeResetToOriginalOnCompletionOfListener($even
     {
         $this->assertEquals('route', $event->getName());
     }
+
+    public function testValidatesBothRouteAndBodyParams()
+    {
+        $services = new ServiceManager();
+        $factory  = new InputFilterFactory();
+
+        $services->setService('FooValidator', $factory->createInputFilter([
+            'from-route' => [
+                'name' => 'from-route',
+                'required' => true,
+                'validators' => [
+                    ['name' => 'Digits'],
+                ],
+            ],
+            'from-body' => [
+                'name' => 'from-body',
+                'required' => true,
+                'validators' => [
+                    ['name' => 'Digits'],
+                ],
+            ],
+        ]));
+        $listener = new ContentValidationListener([
+            'Foo' => ['input_filter' => 'FooValidator'],
+            'validate_route_params' => [
+                'api.rest.test' => ['from-route']
```

Comment:
@weierophinney 
I see the point about the separation of input but imagine this:

A `PATH /user/foo-bar` with a new email address which needs to ensure the uniqueness of the email address.  This means that the user_id from the route needs to be passed in to the InputFilter to filter out the email with the current user.  This PR allows the user_id to come in from the route instead of the consumer needing to pass the user_id along in the body.  

The for idea this option came from https://groups.google.com/a/zend.com/forum/#!topic/apigility-users/9ItFlsDUmtU

I have seen this request a number of time and I wanted to provide a way to define which route params would be injected into the inputfilter instead of injecting all the route params. In the test you can see we are saying that the route param `from-route` needs to be passed into the inputfilter (which technically does not have to validate if the user never defines a validator). 

I can see how the test is confusing because in the test I am validating the route param as the justification that the route param is in fact getting injected into the inputfilter.  I can write a more relevant use-case to avoid the confusion